tchannel/tornado/peer.py

```diff
@@ -618,7 +617,8 @@ def send_with_retry(self, request, peer, retry_limit, connection):
             try:
                 response = yield self._send(connection, request)
                 raise gen.Return(response)
-            except (ProtocolError, TimeoutError) as error:
+            # Why are we retying on all errors????
+            except TChannelError as error:
```

Comment:
What do you mean? This is already a breaking change, what is the update you're talking about? What should `UPGRADE.rst` tell consumers to do?